src/Command/VarnishControllerDeployCommand.php

```diff
@@ -0,0 +1,170 @@
+        }
+        if ($newVclSource === $vclFile->getSource()) {
+            $logger->log('info', 'VCL did not change, will not update');
+        } else {
+            //todo
+            $rollbackID = $client->getHead($vclFile->getId());
+            $logger->log('info', 'Using VCL {id} which head is {rollbackID}', ['id' => $vclFile->getId(), 'rollbackID' => $rollbackID]);
+
+            $vclCommit = $client->updateVCL($vclFile->getId(), $newVclSource);
+            $logger->log('info', 'Pushed new VCL with the commit ID {id}', ['id' => $vclCommit]);
+        }
+
+
+        if (!$groupID) {
+            throw new InvalidOptionException($vclGroup.' needs to be a valid group name');
+        }
+
+        $logger->log(
+            'info', 'Deploying VCL {id} to group {groupID}', ['id' => $vclID, 'groupID' => $groupID]);
+        [$success, $compilationData, $deployData] = $client->deploy($groupID, $vclID);
+        if (!$success) {
+            $logger->log('error', 'Deployment failed');
+            $logger->log('error', 'Compilation errors: {errors}', ['errors' => Json::encode($compilationData)]);
+            $logger->log('error', 'Deployment errors: {errors}', ['errors' => Json::encode($deployData)]);
+
+            $id = $client->rollback($vclID, $rollbackID);
+            $logger->log('info', 'Rolled VCL {vclID} back to {id}', ['vclID' => $vclID, 'id' => $id]);
+
+            return 1;
+        }
+
+        $logger->log('info', 'Successfully deployed VCL {id} to group {groupID}', ['id' => $vclID, 'groupID' => $groupID]);
+        $logger->log('info', 'Compilation data: {data}', ['data' => Json::encode($compilationData)]);
+        $logger->log('info', 'Deployment data: {data}', ['data' => Json::encode($deployData)]);
+
+        return 0;
+    }
+
+    private function getArgumentString(InputInterface $input, string $name): string
+    {
+        $argument = $input->getArgument($name);
+        if (\is_array($argument)) {
+            throw new InvalidArgumentException($name.' needs to be a single value');
+        }
+
+        return (string) $argument;
+    }
+
+    private function requireStringOption(InputInterface $input, string $name): string
+    {
+        $option = $input->getOption($name);
+        if (!$option) {
+            throw new InvalidOptionException($name.' is required');
+        }
+        if (!\is_string($option)) {
+            throw new InvalidOptionException($name.' must be of type string but is '.\gettype($option));
+        }
+
+        return $option;
+    }
+
+    private function convertToBoolean(string $value): bool
+    {
+        switch ($value) {
+            case 'true':
+                return true;
+            case 'false':
+                return false;
+            default:
+                return (bool) $value;
+        }
+    }
+
+    private function readFile(string $fileName): ?string
+    {
+        $content = @file_get_contents($fileName);
```

Comment:
i would prefer an exception over silently ignoring problems with reading the file